backend/users/serializers/user.py

```diff
@@ -40,16 +44,76 @@ def validate(self, attrs):
         return refresh
 
 
+class UserChannelSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = UserChannel
+        fields = ["role", "channel_name", "ui_preferences"]
+
+    def validate_channel_name(self, value):
+        if value not in settings.LEDGER_CHANNELS:
+            raise serializers.ValidationError({"channel": "Channel does not exist"})
+        return value
+
+    def validate_ui_preferences(self, value):
+        if type(value) is not dict:
+            raise serializers.ValidationError({"UI preferences should be a dict"})
```

Comment:
Aren't you missing the `ui_preferences` key in `{"ui_preferences": "UI  preferences should be a dict"}`?

Reply:
To keep consistency yes, key added.